app/uk/gov/hmrc/integrationcatalogue/service/PublishService.scala

```diff
@@ -69,27 +69,42 @@ class PublishService @Inject() (
             case Some(apiDetail: ApiDetail) => apiDetail.apiNumber
             case _ => None
           }
+          maybeExtractedApiNumber = apiNumberExtractor.extract(apiDetailParsed).apiNumber
           maybeApiNumber <- apiNumberGenerator.generate(request.platformType, maybeExistingApiNumber)
           apiDetailWithNumber = maybeApiNumber match {
               // if the generator returns an API number and the extractor finds a number in the title, we use the generator number but also remove the redundant number from the title
               case Some(apiNumber) => apiNumberExtractor.extract(apiDetailParsed).copy(apiNumber = Some(apiNumber))
               case None            => apiNumberExtractor.extract(apiDetailParsed)
           }
+
           maybeTeamId <- maybeExistingIntegrationDetail match {
             case Some(apiDetail: ApiDetail) => Future.successful(apiDetail.teamId)
             case _ => apiTeamsRepository.findByPublisherReference(apiDetailParsed.publisherReference).map {
               case Some(apiTeam) => Some(apiTeam.teamId)
               case _             => None
             }
           }
-          apiDetailWithNumberAndTeam = apiDetailWithNumber.copy(teamId = maybeTeamId)
-          result <- integrationRepository.findAndModify(apiDetailWithNumberAndTeam).map {
+
+          apiDetailWithNumberAndShortDescription = (maybeExtractedApiNumber, apiDetailWithNumber.shortDescription) match {
+            case (Some(parsedApiNumber), Some(shortDesc)) if !shortDesc.endsWith(s"API#$parsedApiNumber") =>
+              apiDetailWithNumber.copy(shortDescription = Some(shortDesc + s" API#$parsedApiNumber"))
+            case (Some(parsedApiNumber), Some(shortDesc)) =>
+              apiDetailWithNumber
+            case (Some(parsedApiNumber), None) =>
+              apiDetailWithNumber.copy(shortDescription = Some(s"API#$parsedApiNumber"))
+            case (None, _) =>
+              apiDetailWithNumber
+          }
```

Comment:
just a small change if preferred
```suggestion
          apiDetailWithNumberAndShortDescription = (maybeExtractedApiNumber, apiDetailWithNumber.shortDescription) match {
            case (Some(parsedApiNumber), Some(shortDesc)) if !shortDesc.endsWith(s"API#$parsedApiNumber") =>
              apiDetailWithNumber.copy(shortDescription = Some(shortDesc + s" API#$parsedApiNumber"))
            case (Some(parsedApiNumber), None) =>
              apiDetailWithNumber.copy(shortDescription = Some(s"API#$parsedApiNumber"))
            case _ => apiDetailWithNumber
          }
```